practical/groovy/crop_rectangle_from_image.groovy

```diff
@@ -0,0 +1,185 @@
+import ome.formats.importer.cli.LoggingImportMonitor
+
+import loci.formats.in.DefaultMetadataOptions
+import loci.formats.in.MetadataLevel
+
+import ij.IJ
+
+// Setup
+// =====
+
+// OMERO Server details
+HOST = "outreach.openmicroscopy.org"
+PORT = 4064
+group_id = "-1"
+// parameters to edit
+image_id = "1001"
+USERNAME = "username"
+PASSWORD = "password"
+
+def connect_to_omero() {
+    "Connect to OMERO"
+
+    credentials = new LoginCredentials()
+    credentials.getServer().setHostname(HOST)
+    credentials.getServer().setPort(PORT)
+    credentials.getUser().setUsername(USERNAME.trim())
+    credentials.getUser().setPassword(PASSWORD.trim())
+    simpleLogger = new SimpleLogger()
+    gateway = new Gateway(simpleLogger)
+    gateway.connect(credentials)
+    return gateway
+
+}
+
+def open_image_plus(HOST, USERNAME, PASSWORD, PORT, group_id, image_id) {
+    "Open the image using the Bio-Formats Importer"
+
+    StringBuffer options = new StringBuffer()
+    options.append("location=[OMERO] open=[omero:server=")
+    options.append(HOST)
+    options.append("\nuser=")
+    options.append(USERNAME.trim())
+    options.append("\nport=")
+    options.append(PORT)
+    options.append("\npass=")
+    options.append(PASSWORD.trim())
+    options.append("\ngroupID=")
+    options.append(group_id)
+    options.append("\niid=")
+    options.append(image_id)
+    options.append("] ")
+    options.append("windowless=true view=Hyperstack ")
+    IJ.runPlugIn("loci.plugins.LociImporter", options.toString())
+}
+
+def find_dataset(gateway, dataset_id) {
+    "Load the Dataset"
+    browse = gateway.getFacility(BrowseFacility)
+    user = gateway.getLoggedInUser()
+    ctx = new SecurityContext(user.getGroupId())
+    return browse.findIObject(ctx, "omero.model.Dataset", dataset_id)
+}
+
+def upload_image(paths, gateway, id) {
+    "Upload an image to OMERO"
+
+    user = gateway.getLoggedInUser()
+    sessionKey = gateway.getSessionId(user)
+
+    config = new ImportConfig()
+    config.debug.set('false')
+    config.hostname.set(HOST)
+    config.sessionKey.set(sessionKey)
+    dataset = find_dataset(gateway, id)
+
+    store = config.createStore()
+    reader = new OMEROWrapper(config)
+
+    library = new ImportLibrary(store, reader)
+    error_handler = new ErrorHandler(config)
+
+    library.addObserver(new LoggingImportMonitor())
+    candidates = new ImportCandidates(reader, paths, error_handler)
+    containers = candidates.getContainers()
+    containers.each() { c ->
+        c.setTarget(dataset)
+    }
+    reader.setMetadataOptions(new DefaultMetadataOptions(MetadataLevel.ALL))
+    return library.importCandidates(config, candidates)
+}
+
+
+// Connect to OMERO
+println "connecting..."
+gateway = connect_to_omero()
+
+println "opening Image..."
+// Open the Image using Bio-Formats
+open_image_plus(HOST, USERNAME, PASSWORD, PORT, group_id, image_id)
+
+// Crop the image
+println "cropping..."
+IJ.makeRectangle(0, 0, 50, 50)
```

Comment:
very small rectangle - it is actually quite hard to believe this is a crop of the image I have passed into the script - maybe we could go for at least twice as large one ?